staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -494,7 +494,7 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return nil, err
 	}
 
-	if err := c.ready.wait(); err != nil {
+	if err := c.ready.wait(ctx); err != nil {
```

Comment:
Ok, so this is governed by this context

https://github.com/kubernetes/kubernetes/blob/10cdaefc1f98edc839d8b9ce6d8ef51745e1466d/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L249-L264